### ISSUE TITLE
Customize pan gesture recognizer sensitivity

### ISSUE THREAD

**Issue by gpapotti** [kukushi/SideMenu#71]
Hi, Kukushi.
Beautiful control, Congrats!
Sometimes pan gesture recognizer is a little too "choosy" in detecting a valid gesture. After a few tests I discovered that it would be useful to add a property in SideMenuController class in order to parametrise the constant value (.25) you're using in isValidateHorizontalMovement. For example on a real iPhone XS you get much better user experience with a value of .45 to guess "horizontality" of the gesture. I'll investigate further because I think it depends on screen dimension/density.

```swift
open class SideMenuController: UIViewController {
...

    private var _horizontalPanSensitivity:CGFloat = 0.25 // .45 was much better on large screens
    
    open var horizontalPanSensitivity: CGFloat
    {
        set
        {
            _horizontalPanSensitivity = newValue
        }
        get
        {
            return _horizontalPanSensitivity
        }
    }
...
}


in extension SideMenuController:
    private func isValidateHorizontalMovement(for velocity: CGPoint) -> Bool {
        if isMenuRevealed {
            return true
        }

        let direction = preferences.basic.direction
        var factor: CGFloat = direction == .left ? 1 : -1
        factor *= shouldReverseDirection ? -1 : 1
        guard velocity.x * factor > 0 else {
            return false
        }
        // Changed: using new sensitivity parameter
        return abs(velocity.y / velocity.x) < horizontalPanSensitivity
    }
```

**Comment by kukushi** [comment]
@gpapotti I have not yet tested it, but fell free to open a PR.

**Comment by madeofmoney** [comment]
Can you please merge this. It's almost impossible to drag the menu out with your finder on iPhone X + models.

**Comment by madeofmoney** [comment]
@kukushi can you please just add this property yourself. We need this desperately for larger phones.

**Comment by madeofmoney** [comment]
It's been 2 years. He isn't going to make a PR.

**Comment by kukushi** [comment]
> It's been 2 years. He isn't going to make a PR.

Shipped in 2.0.7
